### ISSUE TITLE
TestFermiLATBasicImageEstimator.test_run fail

### ISSUE THREAD

**Issue by cdeil** [gammapy/gammapy#1197]
Locally I'm seeing `TestFermiLATBasicImageEstimator.test_run` fail like this:
```
gammapy/maps/hpxnd.py:286: ValueError
```
https://gist.github.com/cdeil/955c58293fb2fc5736865b0a7cf0aecf

It doesn't fail on travis-ci because that test uses an example data file that isn't available here.

I think this is because of a change introduced in #1190 .

Here's a minimal example to reproduce the ValueError:
https://nbviewer.jupyter.org/gist/cdeil/7c1dd02885dfd6b8742e8cae4fec2cdc

@woodmd - I'm confused about when I'm supposed to pass `flat=True` or `flat=False` or how to get this use case to work: given a HPX exposure cube and a WCS geometry with a different energy axis: interpolate in energy and spatially the exposure values and create a 3D WCS exposure map. Is it a bug or am I calling it incorrectly?

**Comment by woodmd** [comment]
For the moment the accessor methods only accept valid pixels so in this example you'll need pass to `flat=True` when you call `get_coords`.    It's on my todo list to make the accessor methods ignore undefined pixels coordinates.

**Comment by cdeil** [comment]
@woodmd - I get the same error
```
ValueError: HPX pixel index out of map bounds.
```
if I change to `flat=True` here:
```
coords = m_wcs.geom.get_coords(flat=True)
```
https://nbviewer.jupyter.org/gist/cdeil/25ff85603a3236bb3355edd53759d964

How can there be undefined "undefined pixel coordinates"?
This is an all-sky HPX exposure image, and the WCS map shouldn't contain any undefined pixels, no?

**Comment by woodmd** [comment]
Yes this is a bug.  It appears the problem is that some pixels in the wcs map are not contained in the energy dimension of the healpix map geometry -- the healpix map starts at 10 GeV but the wcs map starts at 100 MeV.  I think previously it was extrapolating in energy but due to a change I made to the behavior of `coord_to_idx` it's getting an invalid pixel index for the extrapolation.  I'll see if I can put together a PR with a fix shortly.

**Comment by woodmd** [comment]
Ok I've opened a PR that with a bug fix for this issue.  Note that even with this fix your example won't work because you're getting the coordinates from the WCS cube.   You should get the coordinate array from `m_wcs_ref`:
```
coords = m_wcs_ref.geom.get_coords(flat=True)
```
Or alternatively you could do:
```
coords = m_wcs.geom.get_coords()
coords = [coords[0][0,...],coords[1][0,...]]
```
This takes advantage of the fact that the coordinate arrays now have the same dimensionality as the map so using this you can grab a slice of the coordinates for the first image plane.

**Comment by cdeil** [comment]
Looks like this was fixed in #1208 :
```
gammapy/image/tests/test_basic.py::TestFermiLATBasicImageEstimator::test_run PASSED
```

@woodmd - Thanks!
